### ISSUE TITLE
Changes in the INVOICE OPTIONS

### ISSUE THREAD

**Issue by Jorgemgm** [thirtybees/thirtybees#527]
When we change any of the "INVOICE OPTIONS" in "Orders" => "Invoices" the changes change the old Invoices and not only for future.

That hapens with all options, Number of the invoices, "Enable product image", "Legal free text", "Footer text", ....

For instance:
1. If the invoice didn't have and we decide "Add current year to invoice number", the old invoices sent to client without year change and now have year in the invoice number.
2. If we decid "Enable product image" the previous invoices add the product image.
3. If we change the "Legal free text", "Footer text" the previous invoices also change.
...

**Comment by Dh42** [comment]
This is expected behavior in the system.

**Comment by Traumflug** [comment]
I think the problem here is that changing older invoices means that this system can't be used to meet legal requirements regarding archiving.

**Comment by Dh42** [comment]
Legal for where?

**Comment by Traumflug** [comment]
Everywhere where merchants have to keep books or pay taxes.

**Comment by Dh42** [comment]
There was a regression introduced in a previous version that dealt with the product information, that should always be static. As far as the template itself should that be static as well?

**Comment by Traumflug** [comment]
All of the papers created for customer communications should be static. Think of some legal disagreement with a customer, then you have to show up with the quotations and invoices you sent earlier. If these don't match the ones shown by the customer: not good.

Keeping old templates isn't the only way to solve this. Another way would be to collect all the PDFs created in some attic. That's likely the more reliable way. Easy to backup, not dependent on the database.

**Comment by Dh42** [comment]
I am going to close this, unless someone wants to re-open it. I think we missed the easiest solution. There is an option in the back office to save the invoices to disk, so they are static, https://i.imgur.com/sSdtAr9.png

**Comment by Traumflug** [comment]
Yes, this goes into the right direction: automatic storage of a PDF copy on disk.

Unfortunately, this switch no longer works, as the PDF generation library has deprecated it. It's currently a no-op.

**Comment by Traumflug** [comment]
Let's mark this as an enhancement, so it doesn't itch you that much :-)
